vm/devices/storage/disk_nvme/nvme_driver/src/queues.rs

```diff
@@ -125,3 +173,42 @@ fn advance(n: u32, l: u32) -> u32 {
         0
     }
 }
+
+#[derive(Protobuf, Clone, Debug)]
+#[mesh(package = "underhill")]
+pub struct SubmissionQueueSavedState {
+    #[mesh(1)]
+    pub sqid: u16,
+    #[mesh(2)]
+    pub head: u32,
+    #[mesh(3)]
+    pub tail: u32,
+    #[mesh(4)]
+    pub committed_tail: u32,
+    #[mesh(5)]
+    pub len: u32,
+    #[mesh(7)]
+    pub base_mem: u64,
+    #[mesh(8)]
+    pub pfns: Vec<u64>,
```

Comment:
The queues are always linear in physical memory, so we should just need the base gpa here.

Reply:
Yes, agree